accsaber-business/src/main/java/de/ixsen/accsaber/business/RankedMapService.java

```diff
@@ -93,6 +93,27 @@ public void addNewRankedMap(String beatSaverId, Long leaderBoardId, String diffi
         this.scoreRepository.saveAll(nowRankedScores);
     }
 
+    public void removeRankedMap(Long leaderboardId) {
+        Optional<RankedMap> map = this.rankedMapRepository.findById(leaderboardId);
+        if(map.isEmpty()){
+            throw new AccsaberOperationException(ExceptionType.RANKED_MAP_NOT_FOUND,
+                    "The ranked map"  + leaderboardId + " does not currently exist.");
+        }
+
+        Song song = map.get().getSong();
+        List<RankedMap> rankedMaps = song.getRankedMaps();
+        rankedMaps.remove(map);
+        song.setRankedMaps(rankedMaps);
+        this.rankedMapRepository.deleteById(leaderboardId);
```

Comment:
Interestingly, this doesn't delete the map for me. Changing it to this does:
this.rankedMapRepository.delete(map.get())